Duplicati/CommandLine/Commands.cs

```diff
@@ -234,14 +235,86 @@ public static int Affected(TextWriter outwriter, Action<Duplicati.Library.Main.C
             }
         }
 
-        public static List<string> PrefixArgsWithAsterisk(IList<string> argList)
+        /// <summary>
+        /// For bare file names with no wildcards, replace argument with
+        /// the equivalent of prefixing with "*/" so we search all folders.
+        /// </summary>
+        public static IEnumerable<string> PrefixArgsWithAsterisk(IEnumerable<string> argList) =>
+            argList.Select(PrefixArgWithAsterisk);
+
+        /// <summary>
+        /// For bare file names with no wildcards, return argument with
+        /// the equivalent of prefixing with "*/" so we search all folders.
+        /// </summary>
+        private static string PrefixArgWithAsterisk(string arg)
         {
-            // Prefix all filenames with "*/" so we search all folders
-            var specialArgChars = new[] { '*', '?', Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar };
-            return argList.Select(arg => arg.IndexOfAny(specialArgChars) < 0
-                 && !arg.StartsWith("[", StringComparison.Ordinal) ? "*" + Util.DirectorySeparatorString + arg : arg).ToList();
+            var containsSeparators = ContainsDirectorySeparators(arg);
+            var containsWildcards = ContainsWildcards(arg);
+
+            if (!containsSeparators && arg.StartsWith("@", StringComparison.Ordinal))
+            {
+                // Convert to Regexp filter and prefix with ".*/"
+                return $"[.*{Regex.Escape(Util.DirectorySeparatorString + arg.Substring(1))}]";
+            }
+            else if (!containsSeparators && !containsWildcards && !arg.StartsWith("[", StringComparison.Ordinal))
+            {
+                // Prefix files with "*/"
+                return "*" + Util.DirectorySeparatorString + arg;
+            }
+            else
+            {
+                return arg;
+            }
+        }
+
+        /// <summary>
+        /// For folders, replace argument with the equivalent of suffixing
+        /// with "*" so we restore contents in the folder.
+        /// </summary>
+        public static IEnumerable<string> SuffixArgsWithAsterisk(IEnumerable<string> argList) =>
+            argList.Select(SuffixArgWithAsterisk);
+
+        /// <summary>
+        /// For folders, return argument with the equivalent of suffixing
+        /// with "*" so we restore contents in the folder.
+        /// </summary>
+        private static string SuffixArgWithAsterisk(string arg)
+        {
+            var containsWildcards = ContainsWildcards(arg);
+            var endsWithSeparator = arg.EndsWith(Util.DirectorySeparatorString);

```

Comment:
Can we use the `StringComparison.Ordinal` comparison type here?

Reply:
Fixed in latest commit.